### ISSUE TITLE
Add #[allow(dead_code)] and #[allow(non_camel_case_types)] to the generated code

### ISSUE THREAD

**Issue by acertain** [capnproto/capnproto-rust#15]
Generated code produces a large amount of non_camel_case_types and dead_code warnings, it should be marked `allow` so that `rustc` doesn't complain.

**Comment by tv42** [comment]
I've also needed to add `#![allow(unused_qualifications)]` where I have the `include!`, that triggers often.

**Comment by tv42** [comment]
The Rust issue linked has moved to https://github.com/rust-lang/rfcs/issues/752

Since capnproto-rust generates files that mostly has a `pub mod ...` per top-level message and `pub static` for consts, maybe it should just generate those allows for each top-level thing. That doesn't sound too onerous.

**Comment by tv42** [comment]
Updated list of all the lint exceptions I've needed:

```rust
#![allow(clippy::as_conversions)]
#![allow(clippy::exhaustive_enums)]
#![allow(clippy::expl_impl_clone_on_copy)]
#![allow(clippy::match_same_arms)]
#![allow(clippy::module_name_repetitions)]
#![allow(clippy::must_use_candidate)]
#![allow(clippy::renamed_function_params)]
#![allow(clippy::semicolon_if_nothing_returned)]
#![allow(clippy::uninlined_format_args)]
#![allow(clippy::unreadable_literal)]
#![allow(clippy::unwrap_used)]
#![allow(clippy::used_underscore_binding)]
#![allow(dead_code)]
#![allow(unused_qualifications)]
```
